pkg/plugin/provision/spirehelm/spirehelm.go

```diff
@@ -37,63 +62,58 @@ func (h *SpireHelm) Deploy(ctx context.Context, ds plugin.DataSource, kubeCfgFil
 		return fmt.Errorf("no trust zones have been configured")
 	}
 
-	if err := addSPIRERepository(ctx); err != nil {
+	if err := addSPIRERepository(ctx, statusCh); err != nil {
 		return err
 	}
 
-	if err := installSPIREStack(ctx, ds, trustZones); err != nil {
+	if err := installSPIREStack(ctx, ds, trustZones, statusCh); err != nil {
 		return err
 	}
 
-	if err := watchAndConfigure(ctx, ds, trustZones, kubeCfgFile); err != nil {
+	if err := watchAndConfigure(ctx, ds, trustZones, kubeCfgFile, statusCh); err != nil {
 		return err
 	}
 
-	if err := applyPostInstallHelmConfig(ctx, ds, trustZones); err != nil {
+	if err := applyPostInstallHelmConfig(ctx, ds, trustZones, statusCh); err != nil {
 		return err
 	}
 
 	// Wait for spire-server to be ready again.
-	if err := watchAndConfigure(ctx, ds, trustZones, kubeCfgFile); err != nil {
+	if err := watchAndConfigure(ctx, ds, trustZones, kubeCfgFile, statusCh); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (h *SpireHelm) TearDown(ctx context.Context, ds plugin.DataSource) error {
+func tearDown(ctx context.Context, ds plugin.DataSource, statusCh chan<- *provisionpb.Status) error {
 	trustZones, err := ds.ListTrustZones()
 	if err != nil {
 		return err
 	}
 
 	if len(trustZones) == 0 {
-		fmt.Println("no trust zones have been configured")
-		return nil
+		return fmt.Errorf("no trust zones have been configured")
 	}
 
-	if err := uninstallSPIREStack(ctx, trustZones); err != nil {
+	if err := uninstallSPIREStack(ctx, trustZones, statusCh); err != nil {
 		return err
 	}
 	return nil
 }
 
-func addSPIRERepository(ctx context.Context) error {
+func addSPIRERepository(ctx context.Context, statusCh chan<- *provisionpb.Status) error {
 	emptyValues := map[string]interface{}{}
 	prov, err := helm.NewHelmSPIREProvider(ctx, nil, emptyValues, emptyValues)
 	if err != nil {
 		return err
 	}
 
-	statusCh := prov.AddRepository()
-	s := statusspinner.New()
-	if err := s.Watch(statusCh); err != nil {
-		return fmt.Errorf("adding SPIRE Helm repository failed: %w", err)
```

Comment:
This needs a bit of additional thought because it no longer propagates an error at this point.